modules/distribution/resources/api_templates/velocity-template_synapse-config_versioned-webapp.xml

```diff
@@ -0,0 +1,270 @@
+
+#macro ( timeout $config)
+    #if( $config.get("actionSelect") == "neverTimeout" )
+        ## print nothing.
+    #else
+        <timeout>
+            <duration>#print_if_exist( $config.get("actionDuration") 30000)</duration>
+            <responseAction>#print_if_exist( $config.get("actionSelect") "fault")</responseAction>
+        </timeout>
+    #end
+    <suspendOnFailure>
+        <errorCodes>#print_list( $config.get("suspendErrorCode") -1)</errorCodes>
+        <initialDuration>#print_if_exist( $config.get("suspendDuration") 0)</initialDuration>
+        <progressionFactor>#print_if_exist( $config.get("factor") 1.0)</progressionFactor>
+        <maximumDuration>#print_if_exist( $config.get("suspendMaxDuration") 0)</maximumDuration>
+    </suspendOnFailure>
+    <markForSuspension>
+        <errorCodes>#print_list( $config.get("retryErroCode") -1)</errorCodes>
+        <retriesBeforeSuspension>#print_if_exist( $config.get("retryTimeOut") 0)</retriesBeforeSuspension>
+        <retryDelay>#print_if_exist( $config.get("retryDelay") 0)</retryDelay>
+    </markForSuspension>   
+#end
+
+#macro ( http_endpoint $name $ep )
+    <endpoint name="$name">
+        <http uri-template="$util.escapeXml($ep.get("url"))">
+        #timeout( $ep.get('config') )
+        </http>
+    </endpoint>
+#end
+
+#macro ( draw_endpoint $type $endpoint_config )
+
+    #set( $endpointClass = $endpoint_config.get("endpoint_type") )
+    #set( $endpoints = $endpoint_config.get("${type}_endpoints"))
+    #set( $ep_name = "${apiName}_API${type}Endpoint_${resourceNo}")
+
+    ## IF endpoint secured
+    #if($isEndpointSecured)
+        #if($isSecureVaultEnabled)
+            <property xmlns="http://ws.apache.org/ns/synapse"
+                      name="password" expression="wso2:vault-lookup('$securevault_alias')"/>
+            <property xmlns="http://ws.apache.org/ns/synapse"
+                      name="unpw" expression="fn:concat('$username',':',get-property('password'))"/>
+            <property xmlns="http://ws.apache.org/ns/synapse"
+                      name="Authorization"
+                      expression="fn:concat('Basic ', base64Encode(get-property('unpw')))"
+					  scope="transport"/>
+        #else
+            <property xmlns="http://ws.apache.org/ns/synapse"
+                      name="Authorization"
+                      expression="fn:concat('Basic ', base64Encode('$username:$password'))"
+                      scope="transport"/>
+        #end
+    #end
+
+    <send>
+        #if ($endpointClass == "http")
+            #http_endpoint($ep_name $endpoints)
+        #elseif ($endpointClass == "address")
+            <endpoint xmlns="http://ws.apache.org/ns/synapse" name="$ep_name">
+                #set( $advance_ep = $endpoints.get("config") )
+                <address uri="$util.escapeXml($endpoints.get("url"))"
+                    #if($advance_ep.get("format") && ($advance_ep.get("format") != "") && ($advance_ep.get("format") != 'leave-as-is'))
```

Comment:
It i allowed & works without any issue. This code segment was already there, I just formatted it.